autosklearn/pipeline/implementations/CategoryShift.py

```diff
@@ -10,8 +10,16 @@ class CategoryShift(BaseEstimator, TransformerMixin):
 
     def _convert_and_check_X(self, X):
         X_data = X.data if sparse.issparse(X) else X
-        if np.nanmin(X_data) < 0:
-            raise ValueError("X needs to contain only non-negative integers.")
+
+        # Check if data is numeric and positive
+        if not X_data.dtype.kind in set('buif') or \
+            np.nanmin(X_data) < 0:
+            raise ValueError('Categories should be non-negative numbers. NOTE: floats '
```

Comment:
Could you please add a check for the data type, too?

Reply:
Is this really necessary? The check for dtype.kind isn't enough? This code already makes sure that the type is some kind of bool, unsigned, integer or float.

Reply:
Probably not.